src/backend/distributed/commands/index.c

```diff
@@ -628,6 +628,53 @@ PreprocessReindexStmt(Node *node, const char *reindexCommand,
 }
 
 
+/*
+ * ReindexStmtObjectAddress returns list of object addresses in the reindex
+ * statement.
+ */
+List *
+ReindexStmtObjectAddress(Node *stmt, bool missing_ok)
+{
+	ReindexStmt *reindexStatement = castNode(ReindexStmt, stmt);
+
+	List *objectAddresses = NIL;
+
+	if (reindexStatement->relation != NULL)
```

Comment:
this is a large copy&paste from `PreprocessReindexStmt`.

Can we refactor `PreprocessReindexStmt` and move the code that calculates `indOid` to another function, like `Oid ReindexStmtFindIndexOid(ReindexStmt, missingOk)`?

That'd be very useful as it is hard to maintain two copies of these functions.